internal/kafka/internal/handlers/observatorium_proxy.go

```diff
@@ -0,0 +1,54 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/auth"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/handlers"
+	"github.com/gorilla/mux"
+)
+
+type observatoriumProxyHandler struct {
+	clusterService services.ClusterService
+}
+
+func NewObservatoriumProxyHandler(clusterService services.ClusterService) *observatoriumProxyHandler {
+	return &observatoriumProxyHandler{
+		clusterService: clusterService,
+	}
+}
+
+// Delete is the handler for deleting a kafka request
+func (h observatoriumProxyHandler) ValidateTokenAndExternalClusterID(w http.ResponseWriter, r *http.Request) {
+	cfg := &handlers.HandlerConfig{
+		Action: func() (i interface{}, serviceError *errors.ServiceError) {
+			external_id := mux.Vars(r)["cluster_external_id"]
+
+			ctx := r.Context()
+
+			claims, err := auth.GetClaimsFromContext(ctx)
+			if err != nil {
+				return nil, errors.Unauthenticated("unable to authenticate token provided in the request")
+			}
+
+			clientID, err := claims.GetClientId()
+			if err != nil {
+				return nil, errors.Unauthenticated("unable to get client ID from the token")
+			}
+
+			clusters, err := h.clusterService.FindAllClusters(services.FindClusterCriteria{})
+			if err != nil {
+				return nil, errors.GeneralError("failed to validate the request: %v" + err.Error())
+			}
+			for _, cluster := range clusters {
+				if cluster.ExternalID == external_id && cluster.ClientID == clientID {
+					return nil, nil
+				}
+			}
+			return nil, errors.Forbidden("failed to match the client ID in the token against provided external cluster ID: %s" + external_id)
```

Comment:
```suggestion
			if arrays.NoneMatch(clusters, func (cluster *api.Cluster) bool { return cluster.ExternalID == external_id && cluster.ClientID == clientID}) {
				return nil, errors.Forbidden("failed to match the client ID in the token against provided external cluster ID: %s" + external_id)
			}
			
			return nil, nil
```

Reply:
+1 Massimo

My suggestion will be to take `arrays.NoneMatch(clusters, func (cluster *api.Cluster) bool { return cluster.ExternalID == external_id && cluster.ClientID == clientID})` out the if and declare a var that can be used inside the if